opensrp-chw/src/main/java/org/smartregister/chw/activity/KvpPrEPProfileActivity.java

```diff
@@ -67,4 +71,27 @@ protected void setupViews() {
         }
         super.setupViews();
     }
+
+
+    @Override
+    public void refreshMedicalHistory(boolean hasHistory) {
+        Visit lastVisit = getVisit(org.smartregister.chw.util.Constants.Events.KVP_PREP_FOLLOWUP_VISIT);
+        if (lastVisit != null) {
+            rlLastVisit.setVisibility(View.VISIBLE);
+            findViewById(R.id.view_notification_and_referral_row).setVisibility(View.VISIBLE);
+            ((TextView) findViewById(R.id.vViewHistory)).setText(R.string.visits_history_profile_title);
+            ((TextView) findViewById(R.id.ivViewHistoryArrow)).setText(getString(R.string.view_visits_history));
+        } else {
+            rlLastVisit.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public void openMedicalHistory() {
+        KvpPrEPMedicalHistoryActivity.startMe(this, memberObject);
+    }
+
+    private Visit getVisit(String eventType) {
+        return PmtctLibrary.getInstance().visitRepository().getLatestVisit(memberObject.getBaseEntityId(), eventType);
```

Comment:
Is it possible to use the KvpLibrary Instance to access the visitRepository just incase the PmtctLibrary is not initialized since the check for hasPmtct in the ChwApplicationFlavor also stops the initialization of the Pmtct library
```suggestion
        return KvpLibrary.getInstance().visitRepository().getLatestVisit(memberObject.getBaseEntityId(), eventType);
```

Reply:
Cool, let me make the changes